File: hash-table/subarray-sum-equals-k.java
class Solution {
    public int subarraySum(int[] nums, int k) {
        int l = 0, r = 0;
        int sum = 0;
        int count = 0;
        int n = nums.length;

        while (r < n) {
            sum += nums[r];
            while (l <= r && sum > k) {
                sum -= nums[l];
                l++;
            }
            if (sum == k) count++;
            r++;
        }
        return count;
    }
}

File: array/subarray-sum-equals-k.py
class Solution:
    def subarraySum(self, nums: List[int], k: int) -> int:

            count = 0
            curr_sum = 0
            seen = {0:1}

            for i in nums:
                curr_sum+=i
            
                if (curr_sum-k) in seen:
                    count+=seen[curr_sum-k]
                
                if curr_sum in seen:
                    seen[curr_sum]+=1

                else:
                    seen[curr_sum]=1
                    
            return count


       


